### PR TITLE
ServiceAccount : use `InteractiveTable`

### LINKED ISSUE #85000: Update Service Account table to use InteractiveTable component
With the recent updates to the User and Teams list tables, we should look to update the Service Account list table to also use the `InteractiveTable` component

### PR DESCRIPTION
**Pull request by eledobleefe**
**What is this feature?**

We modify the `ServiceAccount` to use the `InteractiveTable`

**Why do we need this feature?**
To be consistent as `OrgUserTable` or `UserTable`already use it.

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #85000

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/serviceaccounts/ServiceAccountTable.tsx [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+  showPaging,
+  totalPages,
+  onChangePage,
+  currentPage,
+  isLoading,
+}: ServiceAccountTableProps) => {
+  const displayRolePicker =
+    contextSrv.hasPermission(AccessControlAction.ActionRolesList) &&
+    contextSrv.hasPermission(AccessControlAction.ActionUserRolesList);
+  /* @ts-ignore */
```
Comment (Clarity-89) [inline public/app/features/serviceaccounts/ServiceAccountTable.tsx:59]:
```suggestion

```

This seems to be not necessary. Also it's better to use `@ts-expect-error`, which will give an error when it's not needed anymore.

File: public/app/features/serviceaccounts/ServiceAccountTable.tsx [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+        header: 'Tokens',
+        cell: ({ cell: { value }, row: { original } }: Cell<'role'>) => {
+          const href = `/org/serviceaccounts/${original.id}`;
+          const ariaLabel = `Edit service account's ${name} details`;
+          return isLoading ? (
+            <Skeleton width={100} />
+          ) : (
+            <Stack alignItems="center">
+              <Icon name="key-skeleton-alt"></Icon>
+              <TextLink href={href} aria-label={ariaLabel} color="primary">
```
Comment (Clarity-89) [inline public/app/features/serviceaccounts/ServiceAccountTable.tsx:161]: I don't think this needs to be a link, the first three columns should be enough.

File: public/app/features/serviceaccounts/ServiceAccountsListPage.tsx [hunk 3]
```diff
@@ -322,52 +292,13 @@ export const ServiceAccountsListPageUnconnected = ({
...
-      flexGrow: 1,
-      color: theme.colors.text.secondary,
-
-      span: {
-        padding: theme.spacing(0.5),
-      },
-    }),
     filterDelimiter: css({
       flexGrow: 1,
     }),
```
Comment (Clarity-89) [inline public/app/features/serviceaccounts/ServiceAccountsListPage.tsx:304]: `pageHeader` and `filterDelimiter` don't seem to be used and you can replace the filter styles with a Box component with a margin to get rid of the styles completely.

**Review by Clarity-89 — COMMENTED** [review]
Looks good! 🎉  Left a few minor comments. Also I think somebody from the Identity access team should give this a look.

**Review by tskarhed — COMMENTED** [review]
Could you extract some of the cell rendering functionality into separate functions? I think that would increase readability.

**Comment by eledobleefe** [comment]
> Looks good! 🎉 Left a few minor comments. Also I think somebody from the Identity access team should give this a look.

Yes, they will be added automatically when the PR turn into a normal one, not a draft. But wanted to check if I was in the right direction 😅

File: public/app/features/serviceaccounts/ServiceAccountTable.tsx (hunk 2 above)
Comment (eledobleefe) [reply public/app/features/serviceaccounts/ServiceAccountTable.tsx:161]: 🤷 I copied it from the original code. It is how it works on `main`. Maybe a question for the **identity team**?

**Review by aarongodin — APPROVED** [review]

File: public/app/features/serviceaccounts/ServiceAccountTable.tsx [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+          </Stack>
+        );
+      default:
+        return (
+          <TextLink href={href} aria-label={ariaLabel} color="primary">
+            {value}
+          </TextLink>
+        );
+    }
+  };
```
Comment (tskarhed) [inline public/app/features/serviceaccounts/ServiceAccountTable.tsx:100]: Does this need to be inside the component body? I was initially thinking that each cell type would have a separate function, but I think this works from a readability point of view, especially with the loading state.

File: public/app/features/serviceaccounts/ServiceAccountTable.tsx [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+      case 'id':
+        return (
+          <TextLink href={href} aria-label={ariaLabel} color="secondary">
+            {original.login}
+          </TextLink>
+        );
+      case 'tokens':
+        return (
+          <Stack alignItems="center">
+            <Icon name="key-skeleton-alt"></Icon>
```
Comment (tskarhed) [inline public/app/features/serviceaccounts/ServiceAccountTable.tsx:87]:
```suggestion
            <Icon name="key-skeleton-alt" />
```

**Review by tskarhed — COMMENTED** [review]
You still need to extract `roles` and `actions` columns. Since you need contextSrv stuff, it might be easier to create the cells in separate function (at least these two).

File: public/app/features/serviceaccounts/ServiceAccountTable.tsx [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+}: ServiceAccountTableProps) => {
+  const columns: Array<Column<ServiceAccountDTO>> = useMemo(
+    () => [
+      {
+        id: 'avatarUrl',
+        header: '',
+        cell: ({ cell: { value }, row: { original } }: Cell<'role'>) => {
+          return getCellContent(value, original, isLoading, 'avatarUrl');
+        },
+        sortType: 'string',
```
Comment (tskarhed) [inline public/app/features/serviceaccounts/ServiceAccountTable.tsx:64]: Do you actually need sort type here? I don't think it makes sense to have sorting enabled for this column

**Review by tskarhed — COMMENTED** [review]
It crashes when I try to sort by ID.
`TypeError: a.split is not a function`

https://github.com/grafana/grafana/assets/1438972/e93d1a94-7997-42a3-a270-8fa548e7f1a0

File: public/app/features/serviceaccounts/ServiceAccountTable.tsx (hunk 6 above)
Comment (eledobleefe) [reply public/app/features/serviceaccounts/ServiceAccountTable.tsx:64]: Nope, I took a look at the original table and it does not sort so I think we could keep just sorting by 'account'. Wdyt?

Comment (tskarhed) [reply public/app/features/serviceaccounts/ServiceAccountTable.tsx:64]: Sounds like an idea :)

Comment (eledobleefe) [reply public/app/features/serviceaccounts/ServiceAccountTable.tsx:64]: Done ;) LGTY?

**Review by tskarhed — APPROVED** [review]
Good job! 👍

**Pull request merged**